### PR TITLE
FIX: Show bookmark options by default when editing

### PR DESCRIPTION
**Pull request by nbianca**
These changes also update user bookmark preferences on the client side
immediately in case user creates more than a bookmark during the same
session.

### CODE REVIEW

**Comment by tgxworld** [comment]
Is there a test we can add here for this change?

File: app/assets/javascripts/discourse/tests/acceptance/bookmarks-test.js [hunk 1]
```diff
@@ -162,6 +167,9 @@ acceptance("Bookmarking", function (needs) {
     await selectKit(".bookmark-option-selector").expand();
     await selectKit(".bookmark-option-selector").selectRowByValue(1);
     await click("#save-bookmark");
+
+    assert.equal(User.current().bookmark_auto_delete_preference, "1");
```
Comment (tgxworld) [inline app/assets/javascripts/discourse/tests/acceptance/bookmarks-test.js:171]: I'm wondering if we need to assert for this when the following assertions after we open the edit bookmark modal is asserting that the options panel is expanded by default.

**Review by tgxworld — APPROVED** [review]
Just one comment but otherwise the code looks good :+1:

File: app/assets/javascripts/discourse/tests/acceptance/bookmarks-test.js (hunk 1 above)
Comment (nbianca) [reply app/assets/javascripts/discourse/tests/acceptance/bookmarks-test.js:171]: I think so. This checks that the user preference was saved for the current user session and the next assert checks if the options panel is expanded only when editing the bookmarks.

**Pull request merged**